### PR TITLE
fix: Anchor links are not wrapping on the sidebar

### LINKED ISSUE #58: Customization via custom components
I think that it's a common requirement to heavily customize the appearance of documentation.
Now I should create a local copy of nextra-theme-docs and customize the style with all the @apply Tailwind directives.
In this way I loose the ability to keep my project updated with nextra-theme-docs new features and the style can become quite messy.

Wouldn't it be better to have nextra-theme-docs more "headless", just creating the structure and accepting as config components for `H1`, `H2`, `SidebarLink` etc?
Since I already have Tailwind in my project, I could create these components already styled and dark-mode ready and pass them to the headless nextra-theme-docs.

What do you think about it?

### LINKED ISSUE #580: Anchor links are not wrapping on the sidebar
CleanShot 2022-07-24 at 16 56 46@2x

### PR DESCRIPTION
**Pull request by promer94**
close #58
Screen Shot 2022-07-25 at 06 42 03
0

### CODE REVIEW

**Comment by B2o5T** [comment]
I tested it on horizontal mode on mobile and expected to see the same title as in breadcrumbs, but it’s different, there is no `oong-` suffix 

09DE3AAB-ED80-4367-99E5-0398284C46E8

File: .changeset/weak-eyes-wonder.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'nextra-theme-docs': patch
+'nextra': patch
```
Comment (B2o5T) [inline .changeset/weak-eyes-wonder.md:3]: It’s change only for docs but not for nextra and blog

File: examples/swr-site/pages/docs/advanced/more/loooooooooooooooooooong-title.en-US.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import Callout from 'nextra-theme-docs/callout'
```
Comment (B2o5T) [inline examples/swr-site/pages/docs/advanced/more/loooooooooooooooooooong-title.en-US.mdx:1]: Can be removed

File: packages/nextra-theme-docs/src/sidebar.tsx [hunk 3]
```diff
@@ -167,12 +168,13 @@ function File({ item, anchors, topLevel }: FileProps) {
       })
 
       return (
-        <li className={active ? 'active' : ''}>
+        <li className={cn(active ? 'active' : '', 'break-words')}>
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/sidebar.tsx:171]:
```suggestion
        <li className={cn(active && 'active', 'break-words')}>
```

File: packages/nextra-theme-docs/src/sidebar.tsx [hunk 4]
```diff
@@ -210,7 +214,7 @@ function File({ item, anchors, topLevel }: FileProps) {
   }
 
   return (
-    <li className={active ? 'active' : ''}>
+    <li className={cn(active ? 'active' : '', 'break-words')}>
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/sidebar.tsx:217]:
```suggestion
    <li className={cn(active && 'active', 'break-words')}>
```

**Comment by promer94** [comment]
Screen Shot 2022-07-25 at 07 17 33

I cant reproduce it. Could you share more infomation ?

**Comment by B2o5T** [comment]
> Screen Shot 2022-07-25 at 07 17 33
> 
> I cant reproduce it. Could you share more infomation ?

Simple try with your mobile with horizontal mode :)

**Comment by B2o5T** [comment]
Also it the same on vertical mode but with zoom 50%
B4C3B7BF-F6E7-40B0-824F-F3AAC463A9E5

**Comment by shuding** [comment]
Oops I missed the comments 😬

We can refine with follow up PRs

**Pull request merged**